enhancements/hypershift/node-lifecycle.md

```diff
@@ -0,0 +1,216 @@
+
+## Glossary
+- MCO - [Machine Config Operator](https://github.com/openshift/machine-config-operator)
+- CAPI - [Cluster API](https://github.com/kubernetes-sigs/cluster-api)
+- Control plane Namespace - The Namespace lifecycled by HyperShift in the management cluster for each guest cluster control plane.
+
+## Motivation
+HyperShift differs from standalone OCP in that the components that run in the Hosted Control Plane (HCP) exists in a logical and physical network different from the guest cluster Nodes.  
+This split enables hiding Node management from the cluster admin end user persona.
+Challenges include but not limited to:
+- Provide the ability to manage Nodes securely in multi-tenancy scenario.
+- Managing compute capacity (i.e. Machines) in a guest cluster from a different management cluster.
+- Managing OS upgrades for a guest cluster from a different management cluster in a centralised fashion.
+- Managing OS config for a guest cluster from a different management cluster in a centralised fashion.
+
+### User Stories
+- As a Service Provider I want to have the ability to manage compute capacity in the provider clusters dynamically via declarative API, so I can satisfy end user workloads compute demand:
+  - Scaling.
+  - Autoscaling.
+  - Autorepair.
+- As a Service Provider I want to have the ability upgrade the OS and Kube version of the Nodes in place for baremetal environments, so Hosts don't need to be setup every time.
+- As a Service Provider I want to have the ability upgrade the OS and Kube version of the Nodes by recreating the instances, so Hosts are recycled regularly for security hygiene.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can setup topologies tolerant to domain failures.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can allocate workloads with concrete architecture or hardware demands.
+- As a Service Provider I want to have the ability to upgrade the control plane independently of the data plane, so they can satisfy different compliance policies.
+- As a cluster Admin I want to have the ability to manipulate Nodes with Kubernetes high level primitives e.g. taints, labels, etc. Without interacting the underlying infra.
+
+### Goals
+- Provide a consumable API for Node management that satisfies HyperShift form factor (multi-tenancy and control plane/data plane decoupling) supporting the following features:
+  - Scaling.
+  - Auto scaling.
+  - Auto repair.
+  - Auto approval.
+  - OS upgrades.
+  - Config updates.
+  - Management/workload cluster network and infra separation.
+  - Multi-tenancy.
+
+### Non-Goals
+
+## Proposal
+### API
+[NodePool](https://github.com/openshift/hypershift/blob/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/api/v1alpha1/nodepool_types.go) is the consumer facing API exposed for Node management.
+This is the consumer facing API for Node management.
```

Comment:
This line is duplicated from the end of the previous line.

```suggestion
```

Reply:
dropped it.